src/cobra/core/gene.py

```diff
@@ -299,3 +294,289 @@ def _repr_html_(self):
             n_reactions=len(self.reactions),
             reactions=format_long_string(", ".join(r.id for r in self.reactions), 200),
         )
+
+
+class GPR(Module):
+    """A Gene Reaction rule in a cobra model, using AST as base class
+
+    Parameters
+    ----------
+    string_gpr : string
+        A GPR in string format, which will be parsed into AST
+    """
+
+    def __init__(self, string_gpr=None, *args, **kwargs):
```

Comment:
So I would try to keep the API consistent. Either have a single constructor (`__init__(self, obj)`) where object is checked against strings, expressions, SBML etc. Or have separate `GPR.from_x` methods and have the default constructor only accept an `ast.Expression` object (so basically just a type conversion). Right now, both strategies are mixed.